File: src/main/java/dev/intermediatebox/list/utils/DoublyLinkedListNode.java
package dev.intermediatebox.list.utils;

public class DoublyLinkedListNode {
  public int value;
  public DoublyLinkedListNode prev;
  public DoublyLinkedListNode next;

  public DoublyLinkedListNode(int value) {
    this.value = value;
  }

  public DoublyLinkedListNode(int value, DoublyLinkedListNode prev, DoublyLinkedListNode next) {
    this.value = value;
    this.prev = prev;
    this.next = next;
  }
}

File: typescript/src/list/util/DoublyLinkedListNode.ts
export default class DoublyLinkedListNode {
  public value: number;
  public prev: DoublyLinkedListNode | null;
  public next: DoublyLinkedListNode | null;

  constructor(value: number) {
    this.value = value;
    this.prev = null;
    this.next = null;
  }

  public DoublyLinkedListNode(value: number, prev: DoublyLinkedListNode, next: DoublyLinkedListNode) {
    this.value = value;
    this.prev = prev;
    this.next = next;
  }
}